gitlab.go

```diff
@@ -214,6 +215,18 @@ func NewClient(token string, options ...ClientOptionFunc) (*Client, error) {
 	return client, nil
 }
 
+// NewJobClient returns a new GitLab API client. To use API methods which require
+// authentication, provide a valid job token.
+func NewJobClient(token string, options ...ClientOptionFunc) (*Client, error) {
+	client, err := newClient(options...)
+	if err != nil {
+		return nil, err
+	}
+	client.authType = jobToken
+	client.token = token
+	return client, nil
+}
```

Comment:
Same about the ordering. Please place this below the `NewBasicAuthClient` function.